extensions/wikia/VideoHandlers/js/handlers/Ooyala.js

```diff
@@ -79,55 +81,104 @@ define('wikia.videohandler.ooyala', [
 			});*/
 		}
 
+		function initRegularPlayer() {
+			log('Create Ooyala player', log.levels.info, logGroup);
+
+			win.OO.Player.create(containerId, params.videoId, createParams);
+		}
+
 		createParams.onCreate = onCreate;
 
 		if (adContext && adContext.getContext().opts.showAds) {
 			if (!dartVideoHelper) {
 				throw 'ext.wikia.adEngine.dartVideoHelper is not defined and it should as we need to display ads';
 			}
 
-			tagUrl = dartVideoHelper.getUrl();
-
 			createParams.vast = {
-				tagUrl: tagUrl
+				tagUrl: dartVideoHelper.getUrl()
 			};
 		}
 
 		// log any errors from failed script loading (VID-976)
-		function loadFail( data ) {
+		function loadFail(data) {
 			var message = data.error + ':';
 
-			$.each( data.resources, function() {
+			$.each(data.resources, function () {
 				message += ' ' + this;
 			});
 
-			log( message, log.levels.error, 'VideoBootstrap' );
+			log(message, log.levels.error, logGroup);
 		}
 
-		// Only load the Ooyala player code once, Ooyala AgeGates will break if we load this asset more than once.
-		if ( window.OO === undefined ) {
-			/* the second file depends on the first file */
-			loader({
+		function initRecoveredPlayer() {
+			log('Create recovered Ooyala player', log.levels.info, logGroup);
+
+			win.googleImaSdkFailedCbList = {
+				originalCbList: [],
```

Comment:
> Are you sure that it is a regular array? Not kind of lazy queue?

Yes, it is a regular array.

> edit: maybe we can clear this array instead of overriding it

Nope, since the callbacks list is filled _and_ executed inside IMA plugin.